util/socksproxy/socksproxy.go

```diff
@@ -46,25 +46,22 @@ func NewSocksProxy(useProxy bool, proxyAddress string) SocksProxy {
 // GetTCPProxyDialer returns a tcp connection. The connection is proxied, if useProxy is true.
 func (socksProxy *SocksProxy) GetTCPProxyDialer() proxy.Dialer {
 	if socksProxy.useProxy {
-		socksProxy.log.WithField("address", socksProxy.proxyAddress).
-			Info("Using proxy connection")
 		// Create a proxy that uses Tor's SocksPort.
 		dialer, err := proxy.SOCKS5("tcp", socksProxy.proxyAddress, nil, nil)
 		if err != nil {
+			socksProxy.log.WithError(err).Error("Failed to create SOCKS5 TCP dialer")
```

Comment:
can do `.Panic()` instead of `.Error()` and remove the panic below.

Reply:
I'd actually prefer to remove all panic from this package... and the others at some point. Not sure why it's even here: `proxyAddress` is user input. It can literally be anything. The app shouldn't crash because of bad user input or broken network.

Or you mean just for the time being, until I get to removing panic. :)

Reply:
> Or you mean just for the time being, until I get to removing panic. :)

This in any case, but:

 > Not sure why it's even here: proxyAddress is user input. It can literally be anything. The app shouldn't crash because of bad user input or broken network.

I dare you to make it crash it bad user input related to socks proxy :P See, this function only returns the dialer, and does not actually dial. Dialing itself will fail if the address is bad.

This is the full SOCKS5 implementation:


```Go
// Copyright 2011 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package proxy

import (
	"context"
	"net"

	"golang.org/x/net/internal/socks"
)

// SOCKS5 returns a Dialer that makes SOCKSv5 connections to the given
// address with an optional username and password.
// See RFC 1928 and RFC 1929.
func SOCKS5(network, address string, auth *Auth, forward Dialer) (Dialer, error) {
	d := socks.NewDialer(network, address)
	if forward != nil {
		if f, ok := forward.(ContextDialer); ok {
			d.ProxyDial = func(ctx context.Context, network string, address string) (net.Conn, error) {
				return f.DialContext(ctx, network, address)
			}
		} else {
			d.ProxyDial = func(ctx context.Context, network string, address string) (net.Conn, error) {
				return dialContext(ctx, forward, network, address)
			}
		}
	}
	if auth != nil {
		up := socks.UsernamePassword{
			Username: auth.User,
			Password: auth.Password,
		}
		d.AuthMethods = []socks.AuthMethod{
			socks.AuthMethodNotRequired,
			socks.AuthMethodUsernamePassword,
		}
		d.Authenticate = up.Authenticate
	}
	return d, nil
}

```

as you can see, `nil` is the only error ever returned.

Of course the implementation might change in this regard, but currently this panic can't happen.

Reply:
> Of course the implementation might change in this regard (seems a bit unlikely though), but currently this panic can't happen.

"... And Other Hilarious Jokes You Can Tell Yourself."

Are you trolling me? :sweat_smile:
That's like going in a direction opposite to reliable software.

Reply:
You know my answer before I give it :D 

Of course I agree that error handling is better and more reliable than a panic. I have seen implementations and assumptions change under my feet as well.

I don't mind the panic in this particular case anyway :woman_shrugging: of course not against removing it anyway :)

Reply:
> can do `.Panic()` instead of `.Error()` and remove the panic below.

Ok done. Will deal with panic later anyway. :)